docs/RFCS/typing.md

```diff
@@ -636,15 +666,15 @@ subsequent step, we check the remaining overload set:
 - If there are no candidates left, type checking fails ("no matching overload").
 - if there is only one candidate left, this is used as the implementation function to use for the call, any
   yet untyped placeholder or constant literal is typed recursively using the type defined by its argument position as desired type,
-  (it is possible to prove, and we could assert here, that the inferred type here is always the desired type)
-  then subsequent steps are skipped.
+  (it is possible to prove, and we could assert here, that the inferred type here is always
```

Comment:
What would you base the proof on? That it's equivalent because we do depth-first typing and types are always fully determined when the recursion returns? Is that true for placeholders?